lib/govuk_publishing_components/presenters/step_by_step_nav_helper.rb

```diff
@@ -92,10 +92,24 @@ def get_list_element(style)
         style == "choice" ? "ul" : "ol"
       end
 
-      def link_href(active, href)
+      def link_href(active, href, parameters = {})
+        external_url = href.start_with?('http')
+        href = external_url ? href : update_href_with_parameters(href, parameters)
         active ? "#content" : href
       end
 
+      def update_href_with_parameters(href, parameters)
```

Comment:
I'd consider renaming this to something a bit more descriptive for readability. Something like `add_step_nav_id_to_query_parameters` or `link_with_step_nav_query_parameter`